AmplifyPlugins/API/AWSAPICategoryPlugin/AWSAPIPlugin+Reachability.swift

```diff
@@ -0,0 +1,83 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+import Foundation
+import Combine
+import Reachability
+
+extension AWSAPIPlugin {
+    @available(iOS 13.0, *)
+    public func reachabilityPublisher() throws -> AnyPublisher<ReachabilityUpdate, Never>? {
+        return try reachabilityPublisher(for: nil)
+    }
+
+    @available(iOS 13.0, *)
+    public func reachabilityPublisher(for apiName: String?) throws -> AnyPublisher<ReachabilityUpdate, Never>? {
+        let hostName = try determineHostName(apiName: apiName)
+        if let networkReachability = self.reachabilityMap[hostName] {
+            return networkReachability.publisher
+        }
+        do {
+            let networkReachability = try NetworkReachabilityNotifier(host: hostName,
+                                                                      allowsCellularAccess: true,
+                                                                      reachabilityFactory: Reachability.self)
+            self.reachabilityMap[hostName] = networkReachability
```

Comment:
nit: remove unnecessary `self`

Reply:
yea, you're right, i was being paranoid here since i got bit by `if let reachabilityMap = self.reachabilityMap` not actually storing an object in the instance map..  but yeah.. will update.